src/main/java/com/diffplug/gradle/spotless/FormatTask.java

```diff
@@ -23,19 +23,36 @@
 import java.util.List;
 import java.util.Locale;
 
+import com.diffplug.common.base.Errors;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.GradleException;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.OutputFiles;
+import org.gradle.api.tasks.SkipWhenEmpty;
 import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.incremental.IncrementalTaskInputs;
 
 public class FormatTask extends DefaultTask {
 	// set by SpotlessExtension, but possibly overridden by FormatExtension
+	@Input
 	public Charset encoding = StandardCharsets.UTF_8;
-	public LineEnding.Policy lineEndingsPolicy = LineEnding.UNIX_POLICY;
+	@Input
+	public LineEnding.Policy lineEndingPolicy = LineEnding.UNIX_POLICY;
+
 	// set by FormatExtension
+	@Input
 	public boolean paddedCell = false;
+	@InputFiles
+	@OutputFiles
+	@SkipWhenEmpty
 	public Iterable<File> target;
+	@Input
```

Comment:
One idea (I think from @oehme in the issue discussion) is looking at the bytecode of the function, but I'm inclined to declare bankruptcy on input properties being clean w.r.t. incremental compilation.  That leaves two options:

A) `spotlessCheck` will be wrong sometimes after you've changed your format
B) Have three tasks: `spotlessCheck/spotlessApply` and `spotlessFastCheck`.  `check/apply` can use the same dumb logic from before, and `fastCheck` will be faster (using incremental stuff) but user beware that it might be wrong if you're using a custom function that you've changed.

I see Spotless' priorities as such:

1. Be correct and repeatable.
2. Make it easy to write a one-liner function to tweak a formatting issue.
3. Be as fast as possible.

Based on these priorities, I'm inclined to go with option B.  It lets us implement `spotlessFastCheck` without making a full correctness guarantee.  We can add a section to the readme `Speeding up Spotless` which lists the `spotlessCheckFast` function and lists any provisos, e.g.

> `spotlessCheckFast` will be incorrect if you just changed the format rules.  But if the rules have not changed, then it will be much faster than plain-old `spotlessCheck`.

If we get it to the point that we don't need an asterisk around correctness, then we can dump the old `spotlessCheck` entirely and promote `spotlessCheckFast` to the sole `check` task.